coriolis/tests/test_data_transfer.py

```diff
@@ -0,0 +1,152 @@
+# Copyright 2023 Cloudbase Solutions Srl
+# All Rights Reserved.
+
+import stat
+import struct
+
+import requests
+import requests_unixsocket
+
+from unittest import mock
+from urllib import parse
+
+from coriolis import data_transfer
+from coriolis import exception
+from coriolis.tests import test_base
+
+
+class DataTranferTestCase(test_base.CoriolisBaseTestCase):
+    """Collection of tests for the Coriolis data transfer module."""
+
+    def setUp(self):
+        super(DataTranferTestCase, self).setUp()
+        self.data_content = 'test-content'.encode()
+        self.fmt = 'gzip'
+        self.mock_session = mock.Mock()
+        self.mock_url = 'http://localhost:8000/'
+        self.mock_response = mock.Mock()
+        self.mock_response.content = self.data_content
+        self.mock_session.post.return_value = self.mock_response
+        self.msg_id = 1
+        self.path = 'test-path'
+        self.offset = 0
+
+    @mock.patch.object(data_transfer, 'CONF')
+    def test_get_session_and_address(self, mock_conf):
+        mock_conf.compressor_address = 'localhost:8000'
+        result = data_transfer._get_session_and_address()
+
+        self.assertIsInstance(result[0], requests.Session)
+        self.assertEqual(result[1], "http://%s/" %
+                         mock_conf.compressor_address)
+
+    @mock.patch.object(data_transfer.os, 'stat')
+    @mock.patch.object(data_transfer, 'CONF')
+    def test_get_session_and_address_unix_socket(self, mock_conf, mock_stat):
+        mock_conf.compressor_address = '/var/run/compressor.sock'
+        mock_stat.return_value.st_mode = stat.S_IFSOCK
+
+        result = data_transfer._get_session_and_address()
+
+        self.assertIsInstance(result[0], requests_unixsocket.Session)
+        self.assertEqual(result[1], "http+unix://%s/" % parse.quote_plus(
+            mock_conf.compressor_address))
+
+    @mock.patch.object(data_transfer.os, 'stat')
+    @mock.patch.object(data_transfer, 'CONF')
+    def test_get_session_and_address_invalid_unix_socket(self, mock_conf,
+                                                         mock_stat):
+        mock_conf.compressor_address = '/var/run/compressor.sock'
+        mock_stat.return_value.st_mode = stat.S_IFREG
+
+        self.assertRaises(exception.CoriolisException,
+                          data_transfer._get_session_and_address)
+
```

Comment:
The following `_get_session_and_address` branches are not explicitly covered: 
https://github.com/cloudbase/coriolis/blob/master/coriolis/data_transfer.py#L41
https://github.com/cloudbase/coriolis/blob/master/coriolis/data_transfer.py#L56

Please add cases for these too

Reply:
The first can be done by literally just calling the function and checking the result is `(None, None)`

The second should be done by NOT mocking `os.stat`, therefore `os.path.exists` will always return `False`, because the mocked `compressor_address` does not in fact exist on the disk